src/csharp/Grpc.Core/Internal/NativeMetadataCredentialsPlugin.cs

```diff
@@ -68,7 +68,8 @@ private void NativeMetadataInterceptorHandler(IntPtr statePtr, IntPtr serviceUrl
             }
             catch (Exception e)
             {
-                Native.grpcsharp_metadata_credentials_notify_from_plugin(callbackPtr, userDataPtr, MetadataArraySafeHandle.Create(Metadata.Empty), StatusCode.Unknown, GetMetadataExceptionStatusMsg);
+                var detail = GetMetadataExceptionStatusMsg + " " + e.ToString();
```

Comment:
nit: it seems not ideal that the same exception handler code appears twice.

I could see a couple of possible cleanups:
a) put this call creds exception handler code into a (small) utility and call that twice
b) looking more closely, I actually think that this exception handler here is almost dead code leftover from [an earlier change that I made :)](https://github.com/grpc/grpc/commit/27670c81a9beec3365ab5b373c707a847c0df503). That is, the only way to get an Exception to fire here that I can think of if is the `QueueUserWorkItem` call threw an exception (which would not happen AFAIK under reasonable circumstances). So it might make sense to 
   * get rid of this exception handler
   * have it attach a different status message to emphasize where it came from

Reply:
Good catch. We need to catch an exception here because throwing when inside a callback from native code can crash the application.

Also, seems like attempting to call `grpcsharp_metadata_credentials_notify_from_plugin` from here is against the C core contract anyway as it should be done from another thread.

Updated the code  to just log an error (which should never happen anyway unless something is horribly wrong). PTAL.